src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -98,19 +98,18 @@ class BoringUtilsException(message: String) extends Exception(message)
   */
 object BoringUtils {
   /* A global namespace for boring ids */
-  private val namespace: SyncVar[Namespace] = new SyncVar
-  namespace.put(Namespace.empty)
+  private[chisel3] case object CacheKey extends BuilderContextCache.Key[Namespace]
+  private def boringNamespace = Builder.contextCache.getOrElseUpdate(CacheKey, Namespace.empty)
 
   /* Get a new name (value) from the namespace */
   private def newName(value: String): String = {
-    val ns = namespace.take()
-    val valuex = ns.name(value)
-    namespace.put(ns)
-    valuex
+    boringNamespace.name(value)
   }
+  /* True if the requested name (value) exists in the namespace */
+  private def checkName(value: String): Boolean = boringNamespace.contains(value)
 
   /* True if the requested name (value) exists in the namespace */
-  private def checkName(value: String): Boolean = namespace.get.contains(value)
+  //  private def checkName(value: String): Boolean = namespace.get.contains(value)
```

Comment:
I think this and the comment above need to be deleted.

Reply:
Done